mythplugins/mythmusic/mythmusic/visualize.cpp

```diff
@@ -530,6 +534,9 @@ bool MonoScope::draw( QPainter *p, const QColor &back )
 ///////////////////////////////////////////////////////////////////////////////
 // WaveForm - see whole track - by twitham@sbcglobal.net, 2023/01
 
+// static class members survive size changes for continuous display
+QImage WaveForm::m_image {nullptr}; // picture of spectrogram
+
```

Comment:
This doesn't handle two frontends playing different songs at the same time. Maybe a hash based on the filename? You'd also need some way to clean up when the player exits.

Reply:
Two frontends are in 2 processes so they have their own independent copies.  This 1 location per process would be an issue only if 1 frontend is displaying 2 songs, right?  I think this is currently impossible.

But since I already save/restore from disk I reverted it back to being destroyed and recreated on every resize.  This reverts to the noticeable blank and redraw time that I was eliminating by keeping the 1 image per process.

Reply:
I think I keyed off the word "static" in the comment. If this is a non-static class member then its one per object and my original comment is wrong.

Reply:
I looked again now that I have more time then this morning. You changed WaveForm::m_image to a static, which means its shared by all instances of the object. You're right though, the there's probably only one object instance per frontend process, and only one frontend process per machine, so in reality it probably doesn't matter.

Reply:
Excellent, I have restored the static background image with an appropriate s_ prefix rather than m_.  A new comment documents how this works for now and could change back to class member if the object quits getting destroyed at resize or more than 1 needs displayed in the future.